conda_mirror/conda_mirror.py

```diff
@@ -18,24 +18,55 @@
 import requests
 import yaml
 
-
 logger = None
 
-
 DEFAULT_BAD_LICENSES = ['agpl', '']
 
-DOWNLOAD_URL="https://anaconda.org/{channel}/{name}/{version}/download/{platform}/{file_name}"
-# The REPODATA template might break in the future if continuum decides to host
-# everything on anaconda/defaults at a different location than all the other
-# channels on anaconda.org
-REPODATA = 'https://conda.anaconda.org/{channel}/{platform}/repodata.json'
 DEFAULT_PLATFORMS = ['linux-64',
                      'linux-32',
                      'osx-64',
                      'win-64',
                      'win-32']
 
 
+def _maybe_split_channel(channel):
+    """Split channel if it is fully qualified. Otherwise default to
+    conda.anaconda.org
+
+    Parameters
+    ----------
+    channel : str
+        channel on anaconda, like "conda-forge" or fully qualified channel like
+        "https://conda.anacocnda.org/conda-forge"
+
+    Returns
+    -------
+    download_template : str
+        defaults to "https://conda.anaconda.org/{channel}/{platform}/{file_name}"
+        The base url will be modified if the `channel` input parameter is
+        fully qualified
+    channel : str
+        The name-only channel. If the channel input param is something like
+        "conda-forge", then "conda-forge" will be returned. If the channel
+        input param is something like "https://repo.continuum.io/pkgs/free/"
+    """
+    # strip trailing slashes
+    channel = channel.strip('/')
+
+    default_url_base = "https://conda.anaconda.org/"
+    url_suffix = "/{channel}/{platform}/{file_name}"
+    if '://' not in channel:
+        # assume we are being given a channel for anaconda.org
+        logger.debug("Assuming %s is an anaconda.org channel", channel)
+        url = default_url_base + url_suffix
+        return url, channel
+    # looks like we are being given a fully qualified channel
+    download_base, channel = channel.rsplit('/', 1)
```

Comment:
Would using any of the urllib.parse functions like urljoin make this more robust?